strawberry/union.py

```diff
@@ -63,6 +63,32 @@ def __hash__(self) -> int:
         # TODO: Is this a bad idea? __eq__ objects are supposed to have the same hash
         return id(self)
 
+    def __or__(self, other: Union[StrawberryType, type]) -> StrawberryType:
+        if hasattr(other, "_type_definition"):
+            return StrawberryUnion(
+                type_annotations=(
+                    *self.type_annotations,
+                    StrawberryAnnotation(other),
+                )
+            )
+
+        if isinstance(other, StrawberryUnion):
+            return StrawberryUnion(
+                type_annotations=tuple(
+                    set(
+                        (
+                            *self.type_annotations,
+                            *other.type_annotations,
+                        )
+                    )
```

Comment:
shall we make type_annotations always a set? 🤔

Reply:
According to [PEP 604](https://www.python.org/dev/peps/pep-0604/):
> The order of the items in the Union should not matter for equality.
> ```
> (int | str) == (str | int)
> (int | str | float) == typing.Union[str, float, int]
> ```
so I think that's a good idea. Only issue could be hashability of the annotations